### DISCUSSION TITLE
Обнулился кэш-файл истории прослушивания Last FM

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#187] [Помощь с ошибкой]
Отсчет прослушанных треков Last FM внезапно начался сначала, сбросилась вся история.
С чем может быть связано и как избежать? Ограничение на размер файла, размер массива?... Почему тогда не началась перезапись старых треков?

![image](https://user-images.githubusercontent.com/76212904/167217859-606ec6d0-e15c-48d2-8b13-d84f8fc10b09.png)
![image](https://user-images.githubusercontent.com/76212904/167217981-0e0fe45d-1d35-4ea4-b354-f1d73e0e23c4.png)

Бэкап тоже завершился ошибкой. Хотя пока не понимаю как он мне поможет восстановить историю
![image](https://user-images.githubusercontent.com/76212904/167218815-c930fe2c-8297-4639-ac6c-704c1ddc6bfa.png)

**Comment by efremych** [comment]
Нашел, кажется, источник беды. В это время я музыку не слушал.
![image](https://user-images.githubusercontent.com/76212904/167220779-17798f35-5bba-4ae9-9850-5602f061d4a8.png)

Номера строк библиотеки, конечно, совпадать не будут с шаблонным кодом.

**upd**: вспомнил/нашел, что была такая [ошибка](https://github.com/Chimildic/goofy/discussions/26)
Историю себе восстановил (правда не через Гугл диск, забыл, что история версий только в web-версии), благо была сохраненная копия плюс-минус свежего файла истории

**Comment by Chimildic** [comment]
Подобная ошибка давно не появлялась. Особенно при чтении файла.
Насколько код далёк от оригинала? Возможно трогаете файл истории на запись где-то вне стандартного алгоритма. Тогда могла нарушится структура файла и как результат ошибка чтения.

**Reply by efremych** [reply]
Ничего криминального… Сам файл в течение вчерашнего дня не трогал (что было раньше, не помню, всякое бывало…). Не могла же структура нарушиться сама по себе. Но первое что бросилось в глаза, конечно, отметка в 13000 треков. Но, наверно, просто совпадение…

![FA9ECBBF-CF85-426C-B5A0-7F1475312478](https://user-images.githubusercontent.com/76212904/167265064-c3d606ee-4983-4f20-ab9a-c65c8bf8b810.jpeg)
![BFD3B924-C3F2-4B04-A16F-74BAFA81814F](https://user-images.githubusercontent.com/76212904/167265067-a292d418-9e9c-4637-ab1b-cad471a9d29f.jpeg)
![458A9A91-8C58-45A7-A8C0-CCF8E39343C0](https://user-images.githubusercontent.com/76212904/167265070-8e0effd4-1635-422e-afe0-00a27781f842.jpeg)
![1B4C91EE-77E3-4A75-BC01-BC164B513CC9](https://user-images.githubusercontent.com/76212904/167265072-af69de2f-2de0-4698-aaf3-4d780504284d.jpeg)
![49A01E7C-C9BD-4AAB-BA34-C57C7C4E2C3B](https://user-images.githubusercontent.com/76212904/167265074-e888fb97-7996-4cb5-851e-fddabfa03891.jpeg)

**Reply by Chimildic** [reply]
Появились какие-то детали для воспроизведения ошибки?

**Reply by efremych** [reply]
Пока не повторялась ошибка.
